apps/desktop/src/app/sync/sync.service.ts

```diff
@@ -72,56 +65,62 @@ export class SyncService {
       instance?: ReplaySubject<ISourceConnector>;
     };
   } = {
-    gdrive: { definition: GDrive, settings: {} },
+    // gdrive: { definition: GDrive, settings: {} },
     onedrive: { definition: OneDriveConnector, settings: {} },
     dropbox: { definition: DropboxConnector, settings: {} },
-    folder: { definition: FolderConnector, settings: {} },
-    s3: { definition: S3Connector, settings: {} },
-    gcs: { definition: GCSConnector, settings: {} },
-    brightcove: { definition: BrightcoveConnector, settings: {} },
+    // folder: { definition: FolderConnector, settings: {} },
+    // s3: { definition: S3Connector, settings: {} },
+    // gcs: { definition: GCSConnector, settings: {} },
+    // brightcove: { definition: BrightcoveConnector, settings: {} },
   };
   destinations: { [id: string]: { definition: DestinationConnectorDefinition; settings: ConnectorSettings } } = {
     nucliacloud: {
       definition: NucliaCloudKB,
       settings: environment.connectors.nucliacloud,
     },
-    // Disable algolia for now
-    // algolia: {
-    //   definition: Algolia,
-    //   settings: {},
-    // },
   };
   sourceObs = new BehaviorSubject(Object.values(this.sources).map((obj) => obj.definition));
-
-  private _queue: Sync[] = [];
-  queue = new ReplaySubject<Sync[]>(1);
-  ready = new Subject<void>();
+  private _syncServer = new BehaviorSubject<string>(localStorage.getItem(SYNC_SERVER_KEY) || '');
+  syncServer = this._syncServer.asObservable();
 
   private _step = new BehaviorSubject<number>(0);
   step = this._step.asObservable();
   private _showSource = new Subject<{ connectorId: string; quickAccessName: string; edit: boolean }>();
   showSource = this._showSource.asObservable();
   private _showFirstStep = new Subject<void>();
   showFirstStep = this._showFirstStep.asObservable();
+  private _isServerDown = new BehaviorSubject<boolean>(true);
+  isServerDown = this._isServerDown.asObservable();
+  private _sourcesCache = new BehaviorSubject<{ [id: string]: Source }>({});
+  sourcesCache = this._sourcesCache.asObservable();
+  currentSource = this.sourcesCache.pipe(map((sources) => sources[this.getCurrentSourceId()]));
 
-  constructor(private sdk: SDKService, private user: UserService) {
-    this.ready.subscribe(() => {
-      this.watchProcessing();
-      this.start();
-    });
+  constructor(private sdk: SDKService, private user: UserService, private http: HttpClient) {
     const account = this.getAccountId();
     if (account) {
       this.setAccount();
     }
-    const queue: Sync[] = JSON.parse(localStorage.getItem(QUEUE_KEY) || '[]');
-    Object.values(queue).forEach((sync) => {
-      if (!sync.completed) {
-        sync.started = false;
-      }
-    });
-    this._queue = queue;
-    this.onQueueUpdate();
-    this.fetchDynamicConnectors();
+    // this.fetchDynamicConnectors();
+    of(true)
+      .pipe(
+        filter(() => !!this._syncServer.getValue()),
```

Comment:
why starting by an `of(true` instead of starting directly by `this._syncServer`?
```suggestion
this._syncServer
      .pipe(
      take(1),
        filter((value) => !!value),
```